sim/src/test/scala/fasedtests/FASEDTestSuite.scala

```diff
@@ -33,24 +64,66 @@ abstract class FASEDTest(
     make((s"run-${backend}%s".format(if (debug) "-debug" else "") +: args):_*)
   }
 
-  def runTest(backend: String, debug: Boolean, args: Seq[String] = Nil, name: String = "pass") = {
-    compileMlSimulator(backend, debug)
+  def runTest(
+      backend: String,
+      debug: Boolean = false,
+      logFile: Option[File] = None,
+      baseRuntimeConfig: RuntimeConfig = baseRuntimeConfig,
+      additionalPlusArgs: Seq[String] = additionalPlusArgs,
+      additionalMakeArgs: Seq[String] = Seq(),
+      behaviorSpec: Option[String] = None) = {
+    val runtimeConfArg: Option[String] = baseRuntimeConfig match {
+      case DefaultRuntimeConfig => None
+      case EmptyRuntimeConfig => Some(s"SIM_RUNTIME_CONF=")
+      case CustomRuntimeConfig(path) => Some(s"SIM_RUNTIME_CONF=$path")
+    }
+    val plusArgs = Seq(s"""EXTRA_SIM_ARGS=${additionalPlusArgs.mkString(" ")}""")
+    val logArg = logFile.map { logName => s"LOGFILE=${logName}" }
+
+    val makeArgs =
+      runtimeConfArg ++:
+      plusArgs ++:
+      logArg ++:
+      additionalMakeArgs
+
+
     if (isCmdAvailable(backend)) {
-      it should s"run on $backend" in {
-        assert(invokeMlSimulator(backend, debug, args) == 0)
+      it should behaviorSpec.getOrElse(s"run on $backend") in {
+        assert(invokeMlSimulator(backend, debug, makeArgs) == 0)
       }
     }
   }
-  def runTests() {
-    runTest("verilator", false)
-    //runTest("vcs", true)
-  }
 
-  behavior of s"FASED Instance configured with ${platformConfigs} driven by target: ${topModuleClass}"
+  compileMlSimulator("verilator", false)
```

Comment:
Why do you need to call compile before running? Shouldn't the make system handle this?

Reply:
It provides a slightly more specific failure report in scalatest, but yeah, the make system could handle this fine.